build/bootstrap/autoshutdown.cron.sh

```diff
@@ -7,65 +7,92 @@
 
 set -euxo pipefail
 
-if [ -z "${1-}" ]; then
-  echo "Usage: $0 <num_periods> [shutdown command...]"
+if [[ -z ${1-} ]]; then
+  echo "Usage: $0 <num_periods>"
   exit 1
 fi
 
 MAX_COUNT="$1"
-shift
 
-if ! [ "$MAX_COUNT" -gt 0 -a "$MAX_COUNT" -lt 10000 ]; then
+export PATH=$PATH:/snap/bin
+
+if ! [[ $MAX_COUNT -gt 0 && $MAX_COUNT -lt 1000 ]]; then
   echo "Invalid argument '$MAX_COUNT'"
   exit 1
 fi
 
 # We maintain the count of how many consecutive iterations of this script did
 # NOT detect a remote session. Once we exceed MAX_COUNT, we shut down.
 # We use /dev/shm which is not persistent over reboots.
-FILE=/dev/shm/autoshutdown-count
+COUNT_FILE=/dev/shm/autoshutdown-count
 COUNT=0
 
 # We ignore active if it has not been modified in the last two
 # days. If one must keep their GCE worker alive longer than that
 # without touching the file in the middle, you can set the mod
 # time on the file into the future. To set a mod time to the future
-# you can use `touch -t [[CC]YY]MMDDhhmm /.active`.
+# you can use `touch -t [[CC]YY]MMDDhhmm /.active` or `touch -d "+3 days" /.active`.
 ACTIVE_FILE=/.active
 AUTO_SHUTDOWN_DURATION=$(( 2 * 24 * 60 * 60 ))
+# warn about /.active every day
+WARN_EVERY=$(( 1 * 24 * 60 * 60 ))
+WARN_URL=https://us-east4-cockroach-workers.cloudfunctions.net/gceworker-autoshutdown
```

Comment:
We should explore putting this behind a vnet so the URL isn't publicly accessible.

Reply:
The URL requires auth (instance-specific tokens), so it's 401/403 by default. I think we can make it even stricter, by using https://cloud.google.com/run/docs/securing/ingress.